File: src/dbgwent-gameplay-client/src/entities/Card.ts
export default class Card {
  id: CardId;
  name: string;
  points: number;
  fraction: string;
  rowType: RowType;
  cardType: CardType;
  abilities: Ability[];
  played: boolean = false;

  constructor(
    id: CardId,
    name: string,
    points: number,
    fraction: string,
    rowType: RowType,
    cardType: CardType,
    abilities: Ability[]
  ) {
    this.id = id;
    this.name = name;
    this.points = points;
    this.fraction = fraction;
    this.rowType = rowType;
    this.cardType = cardType;
    this.abilities = abilities;
  }

  public play(): boolean {
    if (this.played) return false
    this.played = true
    return true
  }
}

export class CardId {
  readonly value: string
  constructor(value: string) {
    this.value = value
  }
}

File: src/db-gwent-server-kotlin/src/main/kotlin/entities/Card.kt
package entities

class Card(
    val id: CardId,
    val type: CardType,
    val fraction: String,
    val abilities: List<Ability> = emptyList(),
    val played: Boolean = false) {

}

data class CardId(val value: String)

enum class CardType {
    Leader,
    Hero,
    Unit,
    Special,
    Weather
}

data class Ability(val value: String)